### DISCUSSION TITLE
Optional ignore GetServerSideProps on client

### DISCUSSION THREAD

**Discussion by jakubriedl** [vercel/next.js#12096] [Help] [answered]
We are currently working on PWA web app with offline support and come to interesting problem about data loading. The app is https://iflix.com quite large catalog of movies and the goal is to load the app immediately without content when service-worker is registered, show some skeleton loading and request only graphql data from server and render on client. While keeping SSR for users that don't have sw active and hit the actual next with request.
**This approach can save precious time when loading page as less data are transferred over network, react will render only once (cause apollo), and hydration won't be needed. That can result to saving several seconds to time to interactive.**

User A (has been on page previously and have active SW)
- opens `/title/movie/123`
- sw serves cached full app with skeleton loading
- some component inside the app loads the data using apollo `useQuery` and re-renders actual with content when done

User B (never been to iflix)
- opens `/title/movie/123`
- and next server serves full page, SSRed, hydrates and then service worker gets registered so next time it is user A

Originally we thought it could be done by `getServerSideProps` but the current behaviour is that if it is hit on client it does the request to the server to get result from this function. A "ignoreGetServerSidePropsOnClient" config that would allow not to request that and pass undefined in props could solve this issue. Doing all of that in `getInitialProps` should be possible but prevents code splitting and other optimizations.

**Comment by timneutkens** [comment, marked as answer]
Feature request is open here: https://github.com/zeit/next.js/issues/11897

**Reply by jakubriedl** [reply]
love it, thank you
